lib/elixir/lib/kernel.ex

```diff
@@ -2125,7 +2125,7 @@ defmodule Kernel do
   an error will be raised when trying to access it next.
   """
   @spec update_in(Access.t(), nonempty_list(term), (term -> term)) :: Access.t()
-  def update_in(data, keys, fun) when is_function(fun, 1) do
+  def update_in(data, [_ | _] = keys, fun) when is_function(fun) do
```

Comment:
@josevalim on the same philosophy, I removed the arity check from here. If I understand correctly, we would do that in this case too.